nixos/modules/misc/nixpkgs-flake.nix

```diff
@@ -106,6 +106,12 @@ in
           [ "nixpkgs=flake:nixpkgs" ]
           ++ lib.optional config.nix.channel.enable "/nix/var/nix/profiles/per-user/root/channels"
         );
+        # Nix will not accept a NIX_PATH with `flake:foo` in it unless flakes
+        # are enabled in the resulting configuration, and this is kind of surprising behaviour.
+        nix.settings.experimental-features = lib.mkDefault [
+          "nix-command"
+          "flakes"
+        ];
```

Comment:
At the very least there should also be an assertion to go with this, as it's quite easy to break

```
nix-repl> testSystem = lib.nixosSystem { modules = [ { nixpkgs.hostPlatform = "x86_64-linux"; nix.settings.experimental-features = [ "auto-allocate-uids" ]; } ]; }

nix-repl> testSystem.config.nix.settings.experimental-features
[ "auto-allocate-uids" ]
```

...Though I am still sorta hesitant to the idea of enabling this by default on all systems that are built in a Flake. It's a bit of a sneaky change and could easily allow behavior that's unwanted on specific systems; maybe add something in the release notes?

Pointing only the channel to a symlink as mentioned above would be preferrable to avoid this entirely, but not a blocker